src/services/llm/ai-sdk-service.ts

```diff
@@ -163,13 +144,18 @@ function selectModel(providerName: string, modelId: ModelName, llmConfig?: LLMCo
       return xai(modelId);
     case "deepseek":
       return (deepseek as (modelId: ModelName) => LanguageModel)(modelId);
-    case "ollama": {
-      const headers = llmConfig?.ollama?.api_key ? { Authorization: `Bearer ${llmConfig.ollama.api_key}` } : {};
-      if (llmConfig?.ollama?.base_url) {
-        const ollamaInstance = createOllama({ baseURL: llmConfig.ollama.base_url || "http://localhost:11434", headers });
-        return ollamaInstance(modelId);
-      }
+    case "ollama":
       return ollama(modelId);
+    case "openrouter": {
+      // Create OpenRouter provider using OpenAI SDK with custom baseURL
+      const openrouter = createOpenAI({
```

Comment:
I'm leaning in favor of using openrouter ai sdk: https://ai-sdk.dev/providers/community-providers/openrouter
Instead. If in the future we update the open ai sdk I don't want it to potentially impact other integrations